sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1062,8 +1069,10 @@ public Instant getWatermark() {
         return initialWatermark;
       }
 
+      //watermark increases as current_timestamp, in case of latency,
+      //developers need to either provide a customized watermarkFn, or tune Triggers.
       return source.spec.getWatermarkFn() != null
-          ? source.spec.getWatermarkFn().apply(curRecord) : curTimestamp;
```

Comment:
Right, I assume it's part of kafka0.10 client support, so users follow this way need to take care of the new scenario. 
`kafka_record_timestamp` is kind of customized `event_timestamp` for me, so I would ask developers to handle late-data properly, mostly with `Trigger.withAllowedLateness()`.